### ISSUE TITLE
Next 13 /app directory error - Functions cannot be passed directly to Client Components because they're not serializable

### ISSUE THREAD

**Issue by Laci556** [clerkinc/javascript#761]
- [x] Review the documentation: https://docs.clerk.dev/
- [x] Search for existing issues: https://github.com/clerkinc/javascript/issues
- [x] Go through package changelog files.
- [ ] Provide the Frontend API key from your application dashboard.

## Package + Version

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`
- [ ] other:

### Version:

```
4.8.0
```

### Browser/OS

Chrome, Next13 app directory project deployed on Vercel

## Description

I created an app following the docs for using Clerk with Next server components. The app deployed fine and I could log in and got redirected to the home page. After refresh however the page crashes and doesn't display anything. This is the error from the Vercel dashboard

```
2023-02-05T14:57:47.184Z	42ab5f16-7291-4f3e-a8f7-401d8efe4d35	ERROR	Error: Functions cannot be passed directly to Client Components because they're not serializable.
  {actor: undefined, sessionClaims: ..., sessionId: ..., session: ..., userId: ..., user: ..., orgId: ..., orgRole: ..., orgSlug: ..., organization: ..., getToken: function, debug: ...}
                                                                                                                                                                    ^^^^^^^^
    at Ma (/var/task/apps/web/.next/server/chunks/3.js:13404:27)
    at Object.toJSON (/var/task/apps/web/.next/server/chunks/3.js:13163:32)
    at stringify (<anonymous>)
    at V (/var/task/apps/web/.next/server/chunks/3.js:13462:53)
    at ping (/var/task/apps/web/.next/server/chunks/3.js:13262:43)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-02-05T14:57:47.185Z	42ab5f16-7291-4f3e-a8f7-401d8efe4d35	ERROR	[Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: '2548614894'
}
```

The root layout is basically the same as the example in the docs and the home page doesn't use any Clerk components or hooks. Client components on another page using `useAuth` from `@clerk/nextjs/app-beta/client` throw the same error.

**Comment by dimkl** [comment]
Hello @Laci556 , thank you for reporting the issue. We have already merged a [fix](https://github.com/clerkinc/javascript/pull/760) and it should be fixed in the next version. Let me check it again with my colleagues and come back to you.

**Comment by dimkl** [comment]
Fix is released with version `@clerk/nextjs @4.8.2`. I will resolve the current issue but please let us know if you have any more issues on that.
